src/main/java/armazem/Armazem.java

```diff
@@ -0,0 +1,83 @@
+package armazem;
+
+import ingredientes.Ingrediente;
+
+import java.util.List;
+import java.util.TreeMap;
+
+public class Armazem {
+    private TreeMap<Ingrediente, Integer> estoque;
+
+    public Armazem(){
+        this.estoque= new TreeMap<>();
+    }
+
+    public void cadastrarIngredienteEmEstoque(Ingrediente ingrediente) {
+        if (!existeIngrediente(ingrediente)) {
+            estoque.put(ingrediente, 0);
+        } else {
+            throw new IllegalArgumentException("Ingrediente já cadastrado.");
+        }
+    }
+
+    public void descadastrarIngredienteEmEstoque(Ingrediente ingrediente) {
+        if (existeIngrediente(ingrediente)) {
+            estoque.remove(ingrediente);
+        } else {
+            throw new IllegalArgumentException("Ingrediente não encontrado");
+        }
+    }
+
+    public void adicionarQuantidadeDoIngredienteEmEstoque(Ingrediente ingrediente, Integer quantidade) {
+        if (existeIngrediente(ingrediente) && quantidadeASerAlteradaIngredienteValida(quantidade)) {
+            int estoqueAtual = consultarQuantidadeDoIngredienteEmEstoque(ingrediente);
+            estoque.put(ingrediente, quantidade + estoqueAtual);
+        } else if (!quantidadeASerAlteradaIngredienteValida(quantidade)) {
+            throw new IllegalArgumentException("Quantidade invalida");
+        } else {
+            throw new IllegalArgumentException("Ingrediente não encontrado");
+        }
+    }
+
+    public void reduzirQuantidadeDoIngredienteEmEstoque(Ingrediente ingrediente, Integer quantidade) {
+        if (quantidade <= 0) {
+            throw new IllegalArgumentException("Quantidade inválida");
+        } else {
+            if (estoque.containsKey(ingrediente)) {
+                var estoqueAtual = consultarQuantidadeDoIngredienteEmEstoque(ingrediente) - quantidade;
+                if (estoqueAtual < 0) {
+                    throw new IllegalArgumentException("Quantidade inválida");
+                } else if (estoqueAtual == 0) {
+                    estoque.remove(ingrediente);
+                } else if (estoqueAtual > 0) {
+                    estoque.replace(ingrediente, estoque.get(ingrediente) - quantidade);
+                }
+            } else {
+                throw new IllegalArgumentException("Ingrediente não encontrado");
+            }
+        }
+
+    }
```

Comment:
nesta opção bastava comparar o estoque com a quantidade a ser reduzida... se for MAIOR, beleza, se não, estoura a exception.